net_transport.go

```diff
@@ -371,7 +376,12 @@ func (n *NetworkTransport) InstallSnapshot(id ServerID, target ServerAddress, ar
 
 	// Set a deadline, scaled by request size
 	if n.timeout > 0 {
-		timeout := n.timeout * time.Duration(args.Size/int64(n.TimeoutScale))
+		transferSize := args.EstimatedSize
+		if args.Size > 0 {
+			transferSize = args.Size
+		}
+
```

Comment:
Would be good to check if both size and estimated size are set to -1 and fail fast if so